std/algorithm/iteration.d

```diff
@@ -879,10 +881,32 @@ template each(alias pred = "a")
         }
         else // if (isForeachBinaryIterable!Iterable)
         {
-            foreach (i, ref e; r)
+            foreach (ref i, ref e; r)
                 cast(void)binaryFun!BinaryArgs(i, e);
         }
     }
+
+    // opApply with >2 parameters. count the delegate args.
+    // only works if it is not templated (otherwise we cannot count the args)
+    void each(Iterable)(Iterable r)
+        if (!isRangeIterable!Iterable && !isForeachIterable!Iterable &&
+            __traits(compiles, Parameters!(Parameters!(r.opApply))))
+    {
+        auto dg(Parameters!(Parameters!(r.opApply)) params) {
```

Comment:
Parameters of opApply is a delegate. I want the parameters of that delegate, or 'the parameters of the parameter'.

On February 11, 2016 4:26:32 PM EST, "H. S. Teoh" <notifications@github.com> wrote:
>>                  cast(void)binaryFun!BinaryArgs(i, e);
>>          }
>>      }
>> +
>> +    // opApply with >2 parameters. count the delegate args.
>> +    // only works if it is not templated (otherwise we cannot count
>the args)
>> +    void each(Iterable)(Iterable r)
>> +        if (!isRangeIterable!Iterable && !isForeachIterable!Iterable
>&&
>> +            __traits(compiles, Parameters!(Parameters!(r.opApply))))
>> +    {
>> +        auto dg(Parameters!(Parameters!(r.opApply)) params) {
>
>Why do you need to apply `Parameters` twice?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/D-Programming-Language/phobos/pull/3837/files#r52669413

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.